stock_prices/stock_prices.py

```diff
@@ -3,8 +3,31 @@
 import argparse
 
 def find_max_profit(prices):
-  pass
+  buy_price = float("inf")
+  sell_price = 0
 
+  if len(prices) == 1: # When there is only one price in prices return 0
+    return 0
+
+  for i, current_price in enumerate(prices):
+    if i == len(prices) - 1:
+      break
+
+    next_price = prices[i + 1]
+
+    if current_price < buy_price: # We want to find the lowest buy price, so if the current price is lower we set it as the new buy price.
+      buy_price = current_price
+      sell_price = next_price # When we found a new low-buy price we reset to the next price
+      print(f"Curr_price: {current_price}, buy_price: {buy_price}")
+      print(f"Next_price: {next_price}, sell_price: {sell_price} \n")
+      continue
+
+    if next_price > sell_price:
+      sell_price = next_price
+      print(f"New next_price: {next_price}, sell_price: {sell_price} \n")
+
+  print(f"Profit: {sell_price - buy_price}  ---------- \n")
+  return sell_price - buy_price
 
```

Comment:
All of the printing you are doing makes great practice especially for debugging. Also I like that you decided to use the enumerate method for stock_prices